orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/VtrList.java

```diff
@@ -0,0 +1,159 @@
+package uk.gov.di.orchestration.shared.entity;
+
+import com.google.gson.annotations.JsonAdapter;
+import net.minidev.json.JSONArray;
+import net.minidev.json.parser.JSONParser;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.orchestration.shared.serialization.VtrListAdapter;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.text.MessageFormat.format;
+import static net.minidev.json.parser.JSONParser.DEFAULT_PERMISSIVE_MODE;
+
+@JsonAdapter(VtrListAdapter.class)
+public class VtrList {
+
+    private static final Logger LOG = LogManager.getLogger(VtrList.class);
+
+    public static final VtrList DEFAULT_VTR_LIST = new VtrList(Collections.emptyList());
+
+    private final List<VectorOfTrust> vtr;
+    private final CredentialTrustLevel credentialTrustLevel;
+    private final List<LevelOfConfidence> levelsOfConfidence;
+    private final VectorOfTrust effectiveVectorOfTrust;
+
+    public VtrList(List<VectorOfTrust> vtr) {
+        validateVtr(vtr);
+        this.vtr = !vtr.isEmpty() ? vtr : List.of(VectorOfTrust.DEFAULT_VECTOR_OF_TRUST);
+        this.credentialTrustLevel =
+                this.vtr.stream()
+                        .map(VectorOfTrust::getCredentialTrustLevel)
+                        .min(Comparator.naturalOrder())
+                        .orElseThrow(); // Should never throw.
+        this.levelsOfConfidence =
+                this.vtr.stream()
+                        .filter(vot -> vot.getCredentialTrustLevel() == this.credentialTrustLevel)
```

Comment:
I don't think it's strictly necessary because currently there's only two Credential Trust Levels. Will remove it for the time being. Because we have to choose the CTL before we go to IPV strictly speaking there could be LoCs we'd have to filter out because they weren't paired with the selected CTL in the VTR, but from speaking o Ethan `C3` and `C4` aren't really on the horizon so no point preempting the behaviour we'd need. Right now if we'e going to IPV the VTR must only contain VoTs with `Cl.Cm`.